src/cli/commands/init.js

```diff
@@ -100,11 +102,25 @@ export async function run(
     }
 
     let answer;
+    let validAnswer = false;
 
     if (yes) {
       answer = def;
-    } else {
-      answer = (await reporter.question(question)) || def;
+    } else { 
+      //loop until a valid answer is provided, if validation is on entry
+      if (entry.validation) {
+        while (!validAnswer) {
+          answer = (await reporter.question(question)) || def;
+          //validate answer
+          if (entry.validation(String(answer))) {
+            validAnswer = true;
+          } else {
+            reporter.error(reporter.lang('invalidPackageName'));
```

Comment:
@kittens I opened a new PR. Issues with my local env. Your suggest changes are in there as well as a cleaner git history. :)